### PR TITLE
YALB-611: Tabs component

### PR DESCRIPTION
**Pull request by ModulesUnraveled**
### Description of work
- Creates the tabs component
- Fixes link underline position (for standard body text links - and supports a custom position for tabs)

### Testing Link(s)
- [ ] Navigate to the [Component Story](https://deploy-preview-106--dev-component-library-twig.netlify.app/?path=/story/molecules-tabs--tabs)

### Functional Review Steps
- [ ] Verify you can install the component with the CLI

### Design Review
- [x] Verify the designs match the [Figma Designs](https://www.figma.com/file/OVfmCW5s1gI7xnrM8ibX2y/UI-Kit-Alpha-%5BYaleSites%5D)
- [x] Review any new [Percy builds](https://percy.io/95144ff9/component-library-twig)

### Accessibility Review
- [ ] Verify the component meets Accessibility requirements

### CODE REVIEW

**Comment by ahitson** [comment]
Desktop looking good, but tabs should be centered.

Is there a way to look at the mobile version? Right now it's creating a bottom scroll bar.

**Review by kara-franco — APPROVED** [review]
Perfection! I really like the Heydon/Inclusive Components implementation, very easy to use with keyboard and/or screen reader. 

I noticed the bar in mobile view as well, I'll be sure to check back in after that is resolved for a final look, but for now, looks awesome!

**Review by kara-franco — CHANGES_REQUESTED** [review]
- [ ] Add `aria-controls` to the tab elements to better associate the tab to the tab panel. (Did some more research about tabs with Nick and adding aria-controls will help VoiceOver and JAWS users). 
Screen Shot 2022-07-12 at 2 45 37 PM

**Comment by miketullo95** [comment]
Will there be a control option for the tabs to have different width lines for the active tab? Currently, the width is fairly large but it would make sense if the site utilizes the thicker divider sizes like 6 or 8.

**Comment by ahitson** [comment]
Much better. 
- Is that the correct icon variation? Seems thick.
- Can we add a color change hover state to the icon?

**Review by kara-franco — CHANGES_REQUESTED** [review]
Very small request after screen reader testing and talking with Nick! 

- [ ] Add `tabindex="0"` to the `<div>` with `role="tabpanel"`

This will help screen reader users get to the panel content when there are no focusable elements in the panel. 

Screen Shot 2022-07-20 at 5 38 26 PM

**Comment by ModulesUnraveled** [comment]
@miketullo95 right now, no. It's thinner on mobile, and thicket on desktop. If we need to make it a configurable thing, we probably could. But since it changes at viewports it would be more difficult. Not off the table, just wanted to reply that no, currently it is not configurable.

@ahitson This is the same icon that's used in the main nav and stuff. HOWEVER, we are still using the thicker versions. We haven't implemented the pro icons yet, which are thinner. I'd like to update all of the icons in a stand-alone PR though so that when we check the visual regression results it'll be easier to verify that the icons are the only thing that change. (review will be easier)

... I think everything else has been addressed!

**Review by ahitson — APPROVED** [review]

**Review by kara-franco — CHANGES_REQUESTED** [review]
@ntay90 I checked the mobile view again and if a mobile user selects a tab that is near an arrow, the tab title remains off screen (I don't remember seeing this before). I would expect that if the tab title is selected, the entire title would move into view? 

**Horizontal (Landscape view) tabbing/switch control on mobile:**
Screen Shot 2022-07-21 at 5 05 25 PM

**Vertical (Portrait view) tabbing/switch control on mobile:** 
Screen Shot 2022-07-21 at 1 28 56 PM

**Expected display when tab title is selected:**
Screen Shot 2022-07-21 at 1 29 06 PM

**Comment by ntay90** [comment]
What do you think of the user's cursor changing to a grab cursor when hovered over the tab bar in overflow? This could indicate that the tab bar can be clicked and dragged. 

Screen Shot of tab bar with grab cursor.

**Comment by ModulesUnraveled** [comment]
@ntay90 Well, a mouse user **can't** actually click and drag. Scroll devices (touch on mobile phones, and laptop trackpads, for example) work, but actually clicking and dragging doesn't scroll the tabs. So I think I would get frustrated if I saw that icon, and wasn't able to perform that action.

For what it's worth, we have a ticket to revisit that functionality (a mouse user clicking and dragging) after doing some user testing, and getting feedback about whether that functionality is a high priority.

**Comment by ModulesUnraveled** [comment]
@kara-franco I think I got the "move focused item into view" (whether by keyboard or mouse) nailed down.

**Review by miketullo95 — APPROVED** [review]

**Review by ntay90 — APPROVED** [review]

**Review by callinmullaney — CHANGES_REQUESTED** [review]
@ModulesUnraveled See code comments for technical suggested changes. 

Lastly, as a mouse user I found the tabs interaction to be a little frustrating with the REALLY LONG tabs navigation. My scroll wheel didn't allow me to quickly go through the list and click-and-drag (as you already mentioned) isn't a thing here. This resulted in me needing to click 20+ times on the arrows to reach the end. Obviously, this is an extreme edge case and the scenario where a tab group even has half as much tabs is probably slim. But it's something to consider.

File: components/01-atoms/controls/base/control.twig [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  }) %}
+  {% set control__element = 'a' %}
+{% endif %}
+
+{% set control__attributes = control__attributes|merge({
+  class: bem(control__base_class, control__modifiers, control__blockname),
+}) %}
+
+<{{ control__element }} {{ add_attributes(control__attributes) }}>
+    {{- control__content -}}
```
[the remark this answers is not in the document]
Comment (ModulesUnraveled) [reply components/01-atoms/controls/base/control.twig:18]: Done

File: components/02-molecules/tabs/_tabs.scss [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+  }
+}
+
+.tabs__nav {
+  @include atoms.list-reset;
+
+  display: flex;
+  gap: var(--size-spacing-7);
+  overflow: auto;
+  scroll-behavior: smooth;
```
[the remark this answers is not in the document]
Comment (ModulesUnraveled) [reply components/02-molecules/tabs/_tabs.scss:78]: Good point!

File: components/02-molecules/tabs/tabs.twig [hunk 3]
```diff
@@ -0,0 +1,71 @@
+{% set tabs__base_class = 'tabs' %}
```
[the remark this answers is not in the document]
Comment (ModulesUnraveled) [reply components/02-molecules/tabs/tabs.twig:11]: I refactored this to use add_attributes a bunch of places, and added a TODO with a link to a JIRA ticket I just created to address refactoring it to use the list and link components.

I'm going to go ahead and merge now, but we can talk more in that ticket, or here, or Slack if you notice something else glaring. Thanks for the review!

**Review by callinmullaney — COMMENTED** [review]
:+1:

**Pull request merged**